src/containers/EmployeeList/index.tsx

```diff
@@ -1,12 +1,97 @@
 import { TO_EMPLOYEES } from '@/configs';
 import ListPageLayout from '@/layout/SubPageLayout';
+import {
+  ChevronIcon,
+  Pagination,
+  Select,
+  Stack,
+  TextInput,
+} from '@mantine/core';
+import { useEffect, useRef, useState } from 'react';
+import styles from './styles.module.scss';
+import IEmployee from '@/types/employee';
+import Employee from './Employee';
 
 interface EmployeeListProps {}
 
 export default function EmployeeList() {
+  const [employeeList, setEmployeeList] = useState([]);
+  const [activePage, setPage] = useState(1);
+  const [search, setSearch] = useState('');
+  const typingTimeOut = useRef<any>(null);
+
+  const handleChangeSearch = (event: any) => {
+    if (typingTimeOut.current) {
+      clearTimeout(typingTimeOut.current);
+    }
+    typingTimeOut.current = setTimeout(() => {
+      setSearch(event.target.value);
+    }, 600);
+  };
+
+  useEffect(() => {
+    fetch('http://localhost:3000/employee')
+      .then((res) => res.json())
+      .then((employees) => {
+        let employeelists = employees.filter((value: IEmployee) =>
+          value.name.toLowerCase().includes(search.trim().toLowerCase()),
+        );
+        setEmployeeList(employeelists);
+      });
+  }, [search]);
+
+  const [loading, setLoading] = useState(false);
+
   return (
     <ListPageLayout rootDir={TO_EMPLOYEES} title="Employees List">
-      <div>EmployeeList</div>
+      <Stack className={styles.mainBox}>
+        <header className={styles.boxHeader}>
+          <div className={styles.containerBoxHeader}>
+            <TextInput
+              ref={typingTimeOut}
+              placeholder="Search"
+              className={styles.searchBox}
+              onChange={(event) => handleChangeSearch(event)}
```

Comment:
Because `onChange` always receives `event` as parameter so this can be shortened to `onChange={handleChangeSearch}`